File: uniffi/src/message/groups/seal_source_couples.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::{Saider, Seqner};
pub use cesrox::message::groups::{
    SealSourceCouples,
    SealSourceCouple,
};

pub fn seal_source_couple_create(seqner: Seqner, saider: Saider) -> SealSourceCouple {
    SealSourceCouple::new(seqner, saider)
}

pub fn seal_source_couples_create(value: Vec<SealSourceCouple>) -> SealSourceCouples {
    SealSourceCouples::new(value)
}

pub fn seal_source_couples_qb64(seal_source_couples: &SealSourceCouples) -> ParsideResult<String> {
    seal_source_couples.qb64()
}

pub fn seal_source_couples_qb64b(seal_source_couples: &SealSourceCouples) -> ParsideResult<Vec<u8>> {
    seal_source_couples.qb64b()
}

pub fn seal_source_couples_qb2(seal_source_couples: &SealSourceCouples) -> ParsideResult<Vec<u8>> {
    seal_source_couples.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/SealSourceCouples.kt
package cesrox.groups

import cesrox.primitives.Seqner
import cesrox.primitives.Saider
import uniffi.cesrox.*
import uniffi.cesrox.SealSourceCouple as SealSourceCoupleNative
import uniffi.cesrox.SealSourceCouples as SealSourceCouplesNative

class SealSourceCouple {
    val value: SealSourceCoupleNative

    constructor(seqner: Seqner, saider: Saider) {
        value = sealSourceCoupleCreate(seqner.value, saider.value)
    }

    constructor(nativeValue: SealSourceCoupleNative) {
        value = nativeValue
    }

    val seqner get() = Seqner(value.seqner)

    val saider get() = Saider(value.saider)
}

class SealSourceCouples {
    val value: SealSourceCouplesNative

    constructor(couples: List<SealSourceCouple>) {
        value = sealSourceCouplesCreate(couples.map { it.value })
    }

    constructor(couples: SealSourceCouplesNative) {
        value = couples
    }

    fun get(index: Int): SealSourceCouple {
        return SealSourceCouple(value.value[index])
    }

    fun qb2(): List<UByte> {
        return sealSourceCouplesQb2(value)
    }
    fun qb64b(): List<UByte> {
        return sealSourceCouplesQb64b(value)
    }
    fun qb64(): String {
        return sealSourceCouplesQb64(value)
    }
}